File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.q = []

    def push(self, x: int) -> None:
        self.q.append(x)

    def pop(self) -> int:
        output = self.q.pop(0)
        return output
    def peek(self) -> int:
        output = self.q[0]
        return output
    def empty(self) -> bool:
        if len(self.q) == 0:
            return True
        return False


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {
    Stack<Integer> inStack = new Stack<>();
    Stack<Integer> outStack = new Stack<>();
    
    public void push(int x) {
        inStack.push(x);
    }
    
    public int pop() {
        while(!inStack.isEmpty()){
            outStack.push(inStack.pop());
        }
        
        int value = outStack.pop();
        
        while(!outStack.isEmpty()){
            inStack.push(outStack.pop());
        }
        
        return value;
    }
    
    public int peek() {
        while(!inStack.isEmpty()){
            outStack.push(inStack.pop());
        }
        
        int value = outStack.peek();
        
        while(!outStack.isEmpty()){
            inStack.push(outStack.pop());
        }
        
        return value;
    }
    
    public boolean empty() {
        return inStack.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */